libtransact/src/contract/archive/mod.rs

```diff
@@ -14,3 +14,586 @@
  * limitations under the License.
  * -----------------------------------------------------------------------------
  */
+
+//! Enables loading smart contract archives from .scar files.
+//!
+//! A .scar file is a tar bzip2 archive that contains two required files:
+//! - A `.wasm` file that contains a WASM smart contract
+//! - A `manifest.yaml` file that contains, at a minimum:
+//!   - The contract `name`
+//!   - The contract `version`
+//!   - The contract's `input` addresses
+//!   - The contract's `output` addresses
+//!
+//! # Examples
+//!
+//! ```no_run
+//! use transact::contract::archive::{default_scar_paths, SmartContractArchive};
+//!
+//! let default_paths = default_scar_paths();
+//! let archive = SmartContractArchive::from_scar_file("xo", "0.1", &default_paths[0])
+//!     .expect("failed to load .scar file");
+//! ```
+
+mod error;
+
+use std::env::{split_paths, var_os};
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+use bzip2::read::BzDecoder;
+use glob::glob;
+use semver::{Version, VersionReq};
+use tar::Archive;
+
+pub use error::Error;
+
+const DEFAULT_SCAR_PATH: &str = "/usr/share/scar";
+const SCAR_PATH_ENV_VAR: &str = "SCAR_PATH";
+
+/// Load default location(s) for .scar files.
+pub fn default_scar_paths() -> Vec<PathBuf> {
```

Comment:
This would more naturally be default_scar_path(); common env variables (PATH, LD_RUN_PATH, etc.) are always singular but also always a list.

Reply:
Updated